l10n_ch_account_statement_base_import/parsers/g11_file_parser.py

```diff
@@ -0,0 +1,236 @@
+            - 'name': string (e.g: 'KBC-INVESTERINGSKREDIET 787-5562831-01')
+            - 'date': date
+            - 'amount': float
+            - 'unique_import_id': string
+            -o 'account_number': string
+                Will be used to find/create the res.partner.bank in odoo
+            -o 'note': string
+            -o 'partner_name': string
+            -o 'ref': string
+
+        :return: a list of transactions
+        :rtype: list
+        """
+
+        transactions = []
+        for line in self.lines[:-1]:
+            if line[0:3] != '097':
+                ref = line[15:42]
+                currency = line[42:45]
+                amount = float(line[45:57]) / 100
+                transaction_date = time.strftime(
+                    '%Y-%m-%d', time.strptime(line[108:116], '%Y%m%d'))
+                # commission = float(line[141:147]) / 100
+                note = ''
+
+                if line[0:3] == '084':
+                    # Fail / Debit record
+                    reject_code = line[128:130]
+                    if reject_code == '02':
+                        # Debit record
+                        amount *= -1
+                        note = self.reject_reason[reject_code]
+                    else:
+                        # Failed transactions. Get the error reason and
+                        # put it on the OBI field.
+                        note = self.reject_reason[
+                            reject_code] + '\n' + _(
+                                "Amount to debit was %s %f") % (
+                                    currency, amount)
+                        amount = 0.0
+
+                # Add information to OBI if the transaction is a test.
+                if line[5] == '3':
+                    note = _("-- Test transaction --") + '\n' + note
+
+                transactions.append({
+                    'name': '/',
+                    'ref': ref,
+                    'unique_import_id': str(uuid.uuid4()),
+                    'amount': amount,
+                    'date': transaction_date,
+                    'note': note,
+                })
+            else:
+                self.balance_end += self._parse_statement_balance_end(line)
+        return transactions
+
+    def validate(self):
+        """Validate the bank statement
+        :param total_line: Last line in the g11 file. Beginning with '097'
+        :return: Boolean
+        """
+
+        total_line = self.lines[-1]
+        transactions = 0
+        transactions += int(
+            total_line[57:69]) + int(
+                total_line[89:101]) + int(
+                    total_line[113:125])
+        return (len(self.statements[0]['transactions']) == transactions)
+
+    def _parse_statement_date(self):
+        """Parse file statement date
+        :return: A date usable by Odoo in write or create dict
+        """
+
+        date = datetime.date.today()
+        return fields.Date.to_string(date)

```

Comment:
You can use fields.Date.today()
https://www.odoo.com/documentation/8.0/reference/orm.html#openerp.fields.Date